### ISSUE TITLE
association and polymorphic 

### ISSUE THREAD

**Issue by AtomGrafiks** [Netflix/fast_jsonapi#37]
Hey, a brief example : 
```ruby

class User < ApplicationRecord
  has_many  :reports, as: :reportable, inverse_of: :reportable
end

class Report < ApplicationRecord
  belongs_to :reportable, polymorphic: true
end

class Group < ApplicationRecord
  has_many  :reports, as: :reportable, inverse_of: :reportable
end
```

serializer : 

```ruby
class ReportSerializer 
  include FastJsonapi::ObjectSerializer

  has_one :reportable
end
```

output : (the type of data not relative to `reportable_type`)
```ruby
:relationships=> {
  :reportable=> {
    :data=> {
      :id=>"bd729369-6ece-454c-8269-437d1e2ed474", 
      :type=>:reportable
    }
  }
}
```
 type is `reportable`, not `user` or `group` and Fast_JSONAPI fetch a `ReportableSerializer` not `UserSerializer` ou `GroupSerializer`

**Implementation of polymorphic association is a great feature** 👍

**Comment by grossadamm** [comment]
We can check that out. I'm curious how that would impact the performance though. Definitely worth investigating, especially for the ease of use.

**Comment by vovimayhem** [comment]
As far as I can see, having a polymorphic `has_many` (or ActiveRecord models using single-table inheritance, which is my case) is a problem right now, as this gem's serializing expects all associated objects to be the same type for a particular relation...

@grossadamm I too wonder how much of a performance hit is to read each associated object's class and infer the serializer used... but we can see it right now with the existing benchmark tests, don't we?

**Comment by vovimayhem** [comment]
...@grossadamm how much performance decrease is the team willing to accept?

**Comment by shishirmk** [comment]
@vovimayhem  We are still fine tuning our benchmark tests but yes as long as its consistently 25 times faster than AMS we are good.

**Comment by shishirmk** [comment]
Released as a part of version 1.1.0
